python/cudf/cudf/tests/test_json.py

```diff
@@ -327,3 +327,25 @@ def test_json_bad_protocol_string():
     got = cudf.read_json(test_string, lines=True)
 
     assert_eq(expect, got)
+
+
+def test_json_corner_case_with_escape_and_double_quote_char(tmpdir):
+    fname = tmpdir.mkdir("gdf_json").join("tmp_json_escape_double_quote")
+
+    pdf = pd.DataFrame(
+        {
+            "a": ['ab"cd', "?1sde", "rdsd"],
+            "b": ["234", "\\", "lstm"],
```

Comment:
Here's a curveball that I don't think is covered with the current code: "\\\" !
So, in theory we need to also track the "escaped" state to make this work for any number of '\'.

Reply:
`"\"` This in file would be written as `"\\"`

Reply:
omg, what happened to my comment. I meant `"\\\"`, or any other number of `'\'` in a row there.

Reply:
:), `"\\\"` in a file would be `"\\\\\\"`

Reply:
But will add a test for that scenario

Reply:
Modified the conditions to check for all these scenarios.

Reply:
@vuule thank you for finding this case and idea for escape_next

Reply:
Wouldn't `"\\\"` in a file be `\"\\\\\\\"`. :rofl:

Reply:
But there was one more scenario `\\\"`

Reply:
But there was one more scenario `\\\"`